stock_analysis_forecast/models/report_stock_forecast.py

```diff
@@ -0,0 +1,136 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Odoo SA <https://www.odoo.com>
+# Copyright 2016 Alex Comba - Agile Business Group
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import fields, models, tools
+
+
+class ReportStockForecast(models.Model):
+
+    _name = 'report.stock.forecast'
+    _description = 'Report Stock Forecast'
+    _auto = False
+
+    date = fields.Date(
+        string='Date')
+    product_id = fields.Many2one(
+        'product.product', string='Product', readonly=True)
+    quantity = fields.Float(readonly=True)
+    incoming_quantity = fields.Float(readonly=True)
+    outgoing_quantity = fields.Float(readonly=True)
+    location_id = fields.Many2one(
+        'stock.location', string='Location', readonly=True)
+    categ_id = fields.Many2one(
+        'product.category', string='Category', readonly=True)
+
+    def init(self, cr):
+        tools.drop_view_if_exists(cr, 'report_stock_forecast')
+        cr.execute("""CREATE OR REPLACE VIEW report_stock_forecast AS (SELECT
```

Comment:
It would be interesting to split this query in several functions that are inheritable, like Odoo does now for all this kind of reports (see https://github.com/odoo/odoo/blob/007607030a2c810dc7b134e926a8203b94babf84/addons/account/report/account_invoice_report.py#L161)